software/Test_Robot_Software/state_machine.py

```diff
@@ -0,0 +1,83 @@
+from robot_state import State
+from api import RobotMovement
+
+import cv2
+
+class StateMachine:
+    """
+    This class is responsible for switching the states
+    of the robot as implementing the state machine logic
+    """
+
+    def __init__(self, robot_movement):
+        self.robot_movement = robot_movement
+        self.state = State.FIND_A_BALL
+        
+        self.max_speed_x = 10
+        self.max_speed_y = 40
+
+        self.FRAME_WIDTH = 848
```

Comment:
This could come from the camera object so you dont have the resolution defined in multiple places